### PR TITLE
Better reporting of anyOf / oneOf schema-validation errors

### PR DESCRIPTION
**Pull request by maddenp-cu**
**Synopsis**

The feedback given to users when schema validation fails and involves an `anyOf` or `oneOf` quantifier is not currently very informative. For example, given a driver schema where the `namelist:` block provides neither `base_file` or `update_values`, the current feedback is e.g.

```
$ uw sfc_climo_gen validate -c sfc_climo_gen.yaml
[2025-08-13T00:39:42]     INFO Validating config against internal schema: sfc-climo-gen
[2025-08-13T00:39:42]    ERROR 1 schema-validation error found in sfc_climo_gen config
[2025-08-13T00:39:42]    ERROR Error at sfc_climo_gen.namelist:
[2025-08-13T00:39:42]    ERROR   {'validate': True} is not valid under any of the given schemas
[2025-08-13T00:39:42]    ERROR YAML validation errors
```

With the updates in this PR:

```
$ uw sfc_climo_gen validate -c sfc_climo_gen.yaml
[2025-08-13T00:40:18]     INFO Validating config against internal schema: sfc-climo-gen
[2025-08-13T00:40:18]    ERROR 1 schema-validation error found in sfc_climo_gen config
[2025-08-13T00:40:18]    ERROR Error at sfc_climo_gen.namelist:
[2025-08-13T00:40:18]    ERROR   {'validate': True} is not valid under any of the given schemas
[2025-08-13T00:40:18]    ERROR   Candidate rules are:
[2025-08-13T00:40:18]    ERROR     'base_file' is a required property
[2025-08-13T00:40:18]    ERROR     'update_values' is a required property
[2025-08-13T00:40:18]    ERROR   At least one must match.
[2025-08-13T00:40:18]    ERROR YAML validation errors
```

Behavior for `oneOf` is similar, but handles an edge case where JSON Schema returns an error object with different properties -- see the unit tests for examples.

**Type**

- [x] Enhancement (adds new functionality)

**Impact**

- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: src/uwtools/tests/config/test_validator.py [hunk 1]
```diff
@@ -167,29 +169,101 @@ def test_config_validator_internal_schema_file():
...
+        assert messages() == "Schema validation succeeded for test"
+        caplog.clear()
+    for bad in [{"color": "blue", "flower": "rose"}]:
+        assert not ok(bad)
+        expected = """
+        1 schema-validation error found in test
+        Error at top level:
+          %s is valid under each of {'required': ['flower']}, {'required': ['color']}
+          Exactly one must match.
+        """
```
Comment (maddenp-cu) [inline src/uwtools/tests/config/test_validator.py:234]: Here's the edge case. The `color` and `flower` properties are mutually exclusive, but when they're _both_ supplied, JSON Schema, instead of reporting `is not valid under...` reports `is valid under each of...`, which is apparently a Bad Thing, though it doesn't sound like it.
